lib/nrpe/client.py

```diff
@@ -0,0 +1,148 @@
+        :param description: A short description of the check
+        :param hostname: Unit hostname. Defaults to a combination of nagios_context and unit name
+        """
+        nagios_context = self.model.config['nagios_context']
+        nagios_servicegroups = self.model.config.get('nagios_servicegroups') or nagios_context
+        unit_name = self.model.unit.name.replace("/", "_")
+        hostname = hostname or f"{nagios_context}-{unit_name}"
+        if not description:
+            description = f'{name} {unit_name}'
+
+        new_check = {
+            'command': command,
+            'description': description,
+            'hostname': hostname,
+            'servicegroup': nagios_servicegroups,
+        }
+
+        if name not in self.state.checks or self.state.checks[name] != new_check:
+            self.state.dirty = True
+            self.state.checks[name] = new_check
+
+    def remove_check(self, name: str):
+        self.state.checks.pop(name, None)
+
+    def commit(self):
+        """Commit checks to NRPE and Nagios"""
+        if not self.state.dirty:
+            logger.info('Skipping NRPE commit as nothing changed')
+            return
+
+        if not self.state.nrpe_ready:
+            logger.info('NRPE relation is not ready')
+            return
+
+        self._write_check_files()
+        self._publish_to_nagios()
+        subprocess.check_call(['systemctl', 'restart', 'nagios-nrpe-server'])
+        self.state.dirty = False
+        logger.info(f'Successfully updated NRPE checks: {", ".join(c for c in self.state.checks)}')
+
+    def _write_check_files(self):
+        """Register the new checks with NRPE and place their configuration files in the appropriate locations"""
+        for check_name in self.state.checks:
+            check = self.state.checks[check_name]
+
+            check_filename = os.path.join(self.nrpe_confdir, f'{check_name}.cfg')
+            check_args = {
+                'check_name': check_name,
+                'command': ' '.join(check['command'])
+            }
+            with open(check_filename, 'w') as check_config:
+                check_config.write(self.check_template % check_args)
+
+            service_filename = os.path.join(self.nagios_exportdir, 'service__{}_{}.cfg'.format(check['hostname'], check_name))
+            service_args = {
+                'hostname': check['hostname'],
+                'description': check['description'],
+                'check_name': check_name,
+                'servicegroup': check['servicegroup']
+            }
+            with open(service_filename, 'w') as service_config:
+                service_config.write(self.service_template % service_args)
+
+    def _publish_to_nagios(self):
+        """Publish check data on the monitors relation"""
+        rel = self.framework.model.get_relation(self._relation_name)
+        rel_data = rel.data[self.model.unit]
+        rel_data['version'] = '0.3'
```

Comment:
What does the version mean? Is it just static or do we have to take it from somewhere?

Reply:
This is the version spec. It's not really used but doesn't hurt to specify

see https://git.launchpad.net/charm-nrpe/tree/README.md#n120